### PR TITLE
Password field

### PR DESCRIPTION
**Pull request by misspia**
BASED ON https://github.com/SnapWars/SnapWar/pull/10

### Features
- Toggles text visibility
- Changes toggle button text based on visibility state ---> will probably make this an icon later
- Hides button when textfield is empty

### CODE REVIEW

File: SnapWar/Components/SWPasswordTextField.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    }
+    
+    fileprivate func setToggleButtonVisiblity() {
+        if let isEmpty = text?.isEmpty {
+            if isEmpty {
+                toggleButton.isHidden = true
+            } else {
+                toggleButton.isHidden = false
+            }
+        } else {
```
Comment (misspia) [inline SnapWar/Components/SWPasswordTextField.swift]: I feel like there's a better way to do this.. please enlighten me

**Comment by davidozhang** [comment]
LOL UWU

File: SnapWar/Components/SWPasswordTextField.swift [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    
+    fileprivate func setToggleButtonText() {
+        if isSecureTextEntry {
+           toggleButton.setTitle("Show", for: .normal)
+       } else {
+           toggleButton.setTitle("Hide", for: .normal)
+       }
+    }
+    
+    fileprivate func setToggleButtonVisiblity() {
```
Comment (davidozhang) [inline SnapWar/Components/SWPasswordTextField.swift]: nit: Visibility

File: SnapWar/Components/SWPasswordTextField.swift [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        
+        toggleButton.setTitleColor(UIColor.black, for: .normal)
+        toggleButton.addTarget(self, action: #selector(toggleTextVisibility), for: .touchUpInside)
+        setToggleButtonText()
+        setToggleButtonVisiblity()
+        
+        rightView = toggleButton
+        rightViewMode = .always
+        
+        addTarget(self, action: #selector(textFieldDidChange), for: .editingChanged)
```
Comment (davidozhang) [inline SnapWar/Components/SWPasswordTextField.swift:31]: the selector should be set by the view controller containing this text field, it should not be defined in the text field class itself

File: SnapWar/Components/SWPasswordTextField.swift (hunk 1 above)
Comment (davidozhang) [reply SnapWar/Components/SWPasswordTextField.swift]: i assume this is logic that is supposed to show the toggle button only when there's text in the text field, here's a simplified code that is logically equivalent:

```
toggleButton.isHidden = true

if let isEmpty = text?.isEmpty {
    toggleButton.isHidden = isEmpty
}
```

**Review by davidozhang — COMMENTED** [review]
good stuff! i left some comments 👍

File: SnapWar/Components/SWPasswordTextField.swift [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    @objc
+    fileprivate func toggleTextVisibility() {
+        isSecureTextEntry = !isSecureTextEntry
+        setToggleButtonText()
+        
+    }
+    
+    @objc
+    func textFieldDidChange() {
+        setToggleButtonVisiblity()
```
Comment (davidozhang) [inline SnapWar/Components/SWPasswordTextField.swift]: should `setToggleButtonText()` also be called here?

File: SnapWar/Components/SWPasswordTextField.swift (hunk 1 above)
Comment (misspia) [reply SnapWar/Components/SWPasswordTextField.swift]: SMART

File: SnapWar/Components/SWPasswordTextField.swift (hunk 4 above)
Comment (misspia) [reply SnapWar/Components/SWPasswordTextField.swift]: Yes, I want to check whether the  toggle button should be shown every time the textfield gets updated

**Pull request merged**